crates/client/db/src/snapshots.rs

```diff
@@ -0,0 +1,55 @@
+use crate::{rocksdb_snapshot::SnapshotWithDBArc, DB};
+use bonsai_trie::id::BasicId;
+use std::{
+    collections::{btree_map, BTreeMap},
+    fmt,
+    sync::{Arc, RwLock},
+};
+
+pub type SnapshotRef = Arc<SnapshotWithDBArc<DB>>;
+
+/// This struct holds the snapshots. To avoid holding the lock the entire time the snapshot is used, it's behind
+/// an Arc. Getting a snapshot only holds the lock for the time of cloning the Arc.
+pub struct Snapshots {
+    db: Arc<DB>,
+    snapshots: RwLock<BTreeMap<BasicId, SnapshotRef>>,
+    max_saved_snapshots: Option<usize>,
```

Comment:
Why "saved"? They're not persisted, right?

Reply:
If I'm not mistaken we keep a certain amount of snaphots in RAM to be able to fetch historical values much faster than with a simple iterative lookup from the tip of the chain. These are lost on node restart though.